lib/rake/test.rake

```diff
@@ -64,6 +67,13 @@ namespace :test do
     RakeUtils.wait_for_url CDO.studio_url('', CDO.default_scheme)
   end
 
+  task :mark_dtt_green do
+    test_branch_sha = GitHub.sha('test')
```

Comment:
I think this might be a mistake, here and in our DOTD script as well.  This helper actually uses Octokit to ask Github for the sha of the `test` branch at the end of the build, but if somebody pushed to the test branch while the build was running I think we could end up with a different sha than the one we actually ran tests against.  We may want to ask for the sha of the local test branch instead, or capture this at the start of the build.

Reply:
Great catch Brad! This is a mistake here. I've fixed it by looking at the HEAD commit on the test machine. I believe this is sufficient because if the test machine bits have changed mid-DTT, then we don't even know if we tested the old or new bits, and marking the original HEAD commit green wouldn't be any more correct.

As for the bin/dotd script, the change I made here wouldn't be very helpful to make there, because by the time the dotd has the opportunity to mark the DTT green, any changes to the test branch would likely have already been pulled down to the test machine. In that case remembering the HEAD commit from when the DTT was initiated could still save our butts in some cases, and is probably a good idea for us to implement. I'm going to say that's out of scope for this PR, though.